app/src/main/java/chat/rocket/android/authentication/signup/presentation/SignupPresenter.kt

```diff
@@ -3,20 +3,26 @@ package chat.rocket.android.authentication.signup.presentation
 import chat.rocket.android.authentication.presentation.AuthenticationNavigator
 import chat.rocket.android.core.lifecycle.CancelStrategy
 import chat.rocket.android.helper.NetworkHelper
+import chat.rocket.android.server.domain.GetCurrentServerInteractor
+import chat.rocket.android.server.infraestructure.RocketChatClientFactory
 import chat.rocket.android.util.launchUI
 import chat.rocket.common.RocketChatException
-import chat.rocket.core.RocketChatClient
 import chat.rocket.core.internal.rest.login
 import chat.rocket.core.internal.rest.signup
 import javax.inject.Inject
 
 class SignupPresenter @Inject constructor(private val view: SignupView,
                                           private val strategy: CancelStrategy,
-                                          private val navigator: AuthenticationNavigator) {
-    @Inject lateinit var client: RocketChatClient
+                                          private val navigator: AuthenticationNavigator,
+                                          private val serverInteractor: GetCurrentServerInteractor,
+                                          private val factory: RocketChatClientFactory) {
 
     fun signup(name: String, username: String, password: String, email: String) {
+        val server = serverInteractor.get()
         when {
+            server == null -> {
```

Comment:
Why would we have a null server at this point? When the user comes to Sign Up's screen means that the user has already set up the server before.